docs/polymorfisme.md

```diff
@@ -0,0 +1,257 @@
+- er moet een basis type worden gedefinieerd voor de vier datum types. Deze basis type kan dan worden gebruikt als type voor een datum veld. De vier datum types gebruiken de **allOf** element om de basis type te overerven
+- de **discriminator** element moet worden gebruikt om aan te geven welk veld moet worden gebruikt om de type van het datum veld te bepalen. De **discriminator** en de bijbehorende type worden in de basis type gedefinieerd zodat ze door alle vier datum types worden overgeërfd.
+
+Na het toepassen van polymorfisme zien de datum types er als volgt uit:
+
+```yaml
+AbstractDatum:
+  type: object
+  properties:
+    type: string
+  required:
+    - type
+  discriminator:
+    propertyName: type
+  mapping:
+    Datum: '#/components/schemas/VolledigeDatum'
+    OnbekendDatum: '#/components/schemas/OnbekendDatum'
+    JaarDatum: '#/components/schemas/JaarDatum'
+    JaarMaandDatum: '#/components/schemas/JaarMaandDatum'
+VolledigeDatum:
+  allOf:
+    - $ref: '#/components/schemas/AbstractDatum'
+    - type: object
+      properties:
+        datum:
+          type: string
+          format: date
+  required:
+    - datum
+JaarMaandDatum:
+  allOf:
+    - $ref: '#/components/schemas/AbstractDatum'
+    - type: object
+      properties:
+        jaar:
+          type: integer
+          maximum: 9999
+        maand:
+          type: integer
+          minimum: 1
+          maximum: 12
+  required:
+    - jaar
+    - maand
+JaarDatum:
+  allOf:
+    - $ref: '#/components/schemas/AbstractDatum'
+    - type: object
+      properties:
+        jaar:
+          type: integer
+          maximum: 9999
+  required:
+    - jaar
+OnbekendDatum:
+  allOf:
+    - $ref: '#/components/schemas/AbstractDatum'
+    - type: object
+      properties:
+        indicatieOnbekend:
+          type: bool
+  required:
+    - indicatieOnbekend
+```
+
+In de nieuwe datum definities zijn de type enum waarden verwijderd. Deze zijn niet meer nodig omdat standaard de naam van de afgeleide schemas (in dit geval: VolledigeDatum,JaarMaandDatum,JaarDatum,OnbekendDatum) als enum waarde wordt gebruikt. Van deze standaard 'mapping' kan worden afgeweken door gebruik te maken van het **mapping** element van de discriminator. In bovenstaande definities is gebruik gemaakt van het **mapping** element omdat 'Datum' als type waarde is gekozen voor een VolledigeDatum type.
+
+Het toepassen van polymorfisme is niet zonder consequenties. De belangrijkste nadelen voor het toepassen van polymorfisme zijn:
+- de definitie wordt complexer. In plaats van één datum type zijn er nu vijf datum types. Voor een datum type met slechts vier velden kun je je afvragen of het handig is polymorfisme toe te passen. Voor de Verblijfplaats type is het wel handig omdat elk Verblijfplaats type meerdere velden heeft die niet bij een andere Verblijfplaats type voorkomen. Ook is de logica om te bepalen om welk Verblijfplaats type het gaat veel complexer.
```

Comment:
bij datumtype: handig misschien in absoluut noodzakelijk veranderen?
Bij verblijfplaats: wel handig veranderen in "biedt het zeker meerwaarde"?

Reply:
En misschien de argumenten daarbij noemen. Misschien niet absloluut noodzakelijk omdat het maar 4 velden zijn, maar als validatie op het typoe belangrijk is (en dat is het met een datum wel lijkt mij), dan is dit d eoplossing om dat te kunenn afdwingen.

Reply:
done